README.md

```diff
@@ -200,6 +200,28 @@ Then you can simply point the converted model path in the config file by changin
 
 For further information, please refer to [#15](https://github.com/facebookresearch/maskrcnn-benchmark/issues/15).
 
+To finetune the pretrained maskRCNN to train fasterRCNN or maskRCNN on your own dataset, four steps are needed: <br />
+
+**step 1**. Run the demo.py to test your package is installed successfully. Meanwhile, you can save the pretrained model, which has been converted from pkl format to pth format, which can be loaded by a pytorch model. The model is a data of COCODEmo object in the demo, so you can save the weights of the pretrained model as follow: 
+```python
+    pretrained_model_path = "./pretrained_model/maskRCNN.pth"
+    check_point = {'state': coco_demo.model.state_dict()}
+    torch.save(check_point, pretrained_model_path)
+```
+This way, no need to look into [_load_file fun](https://github.com/facebookresearch/maskrcnn-benchmark/blob/f25c6cff92d32d92abe8965d68401004e90c8bee/maskrcnn_benchmark/utils/checkpoint.py#L117). BTW, in case you want to look into this: the [state_dict](https://github.com/pytorch/pytorch/blob/828cb18fa35c7c132ab920de1c0dc6d859f152d6/torch/nn/modules/module.py#L602) method of a Module object saves the weights in an OrderedDict. <br />
+
+**step 2**. Create a fastercnn or maskRCNN model using a yaml file in the configs folder. When doing this, you need to call the cfg.merge_from_list method to change the [_C.MODEL.ROI_BOX_HEAD.NUM_CLASSES =81 ](https://github.com/facebookresearch/maskrcnn-benchmark/blob/f25c6cff92d32d92abe8965d68401004e90c8bee/maskrcnn_benchmark/config/defaults.py#L182) to the number of classes in your dataset (background does not count in this implementation).
+
+**step 3**. Two classes are needed. The 1st class to create fetches one image and its corresponding bbox and label without bundling the bbox and label in a BoxList object. This class is a counterpart of the parent class of the [COCODataset](https://github.com/facebookresearch/maskrcnn-benchmark/blob/f25c6cff92d32d92abe8965d68401004e90c8bee/maskrcnn_benchmark/data/datasets/coco.py#L9).  In each iteration, this class generates a tuple (img, bbox, label), in which img is generated by Image.open(...).convert('RGB'); bbox is a 2d numpy array and label is 1d array (mask information if it exists). The 2nd class inherits from the 1st class and the major responsibility of this class is to wrap the truths of an image in a BoxList object, similar to this [line](https://github.com/facebookresearch/maskrcnn-benchmark/blob/f25c6cff92d32d92abe8965d68401004e90c8bee/maskrcnn_benchmark/data/datasets/coco.py#L43).
```

Comment:
Note that you actually only need a single class, and it's not necessary to write two classes.
See for example [the Pascal VOC dataset](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/data/datasets/voc.py).
One drawback though is that for now, evaluation is dependent on the dataset (as it calls some internal attributes of the dataset). This should be improved in the future.

Reply:
Yes. A single class, which fetches the bbox, labels and wrap them in a Boxlist object, is fine. I separated fetching and wrapping. I am not sure which design is easier to follow as I am already familiar with this project.

Reply:
I see what you mean. The parental class doesn't have to be COCODataset but a Class that opens an image and fetches bbox, labels and segmentation (if any).  Building a customized dataset which is compatible with the maskRCNN implementation will depend on how the user's data look. If no class like the "torchvision.datasets.coco.CocoDetection" exists, it is more clear to build one class. I think this might be more common in a real application rather than a benchmark dataset. 

The maskRCNN supports a customized dataset that returns a tuple (image, target, idx) by the __getitem__ method.  I think what the __getitem__ method returns is the only thing that a user should consider, is it right? Do you think whether we need to write these points in the "Adding your own dataset" section of README?

Reply:
> The maskRCNN supports a customized dataset that returns a tuple (image, target, idx) by the getitem method. I think what the getitem method returns is the only thing that a user should consider, is it right? Do you think whether we need to write these points in the "Adding your own dataset" section of README?

Yes, good point, that should be more explicitly mentioned in the `Adding your own dataset` section of the README.